pkg/notification/manager.go

```diff
@@ -66,6 +72,8 @@ func (am *AtomicManager) Remove(id int) {
 
 // Send sends the notification to the registered handlers
 func (am *AtomicManager) Send(notification interface{}) {
+	am.lock.Lock()
```

Comment:
If the user calls `Remove` in their notification callback, does that cause deadlock?

Reply:
How user will call `Remove` in the callback? I think it will be `race` exception?
I can try a sample code, and can check.